File: sorting/insertion-sort/python/insertion.py
#Time-Complexity=O(n^2)
array=[5,3,2,1,4]

n=len(array)

for i in range (1,n):
    insert_ind=i
    current_val=array.pop(i)
    for j in range (i-1,-1,-1):
        if array[j]>current_val:
            insert_ind=j
    array.insert(insert_ind,current_val)

print("Sorted array",array)

File: sorting/insertion-sort/C++/insertion.cpp
#include <iostream>
int main(){
    int array[]={5,4,6,3,2};
    int n= sizeof(array)/sizeof(array[0]);
    
    // printf("%d",n);
    
    for(int i=1;i<n;i++){
        int insert_ind=i;
        int current_val=array[i];
        int j = i-1;
        while(j>=0&&array[j]>current_val){
            array[j+1]=array[j];
            j=j-1;
        }
        array[j + 1] = current_val;
        
    }
    
    for(int i=0;i<n;i++){
        std::cout<<array[i]<<" ";
    }
    
}